src/arch/x86_64.rs

```diff
@@ -90,6 +114,8 @@ impl hal_core::interrupt::Handlers for InterruptHandlers {
     where
         C: hal_core::interrupt::ctx::Context + hal_core::interrupt::ctx::CodeFault,
     {
+        // Safety: we are crashing. Don't worry about it.
+        unsafe { force_unlock_writers() }
         tracing::error!(registers = ?cx.registers(), "double fault",);
         oops(&format_args!("  DOUBLE FAULT\n\n{}", cx.registers()))
```

Comment:
I feel like we should probably do the `force_unlock_writers` calls within `oops`, and not log anything directly from within these handlers. Perhaps we should add an `interrupt: Option<&dyn Context>` argument to `oops`?

Reply:
> Perhaps we should add an `interrupt: Option<&dyn Context>` argument to `oops`?

slapping myself for not thinking about that, god.

Reply:
@mystor changed it to work that way in 2f213a5d4e87a4e40c138a228994402a677bef0c. Ended up needing a somewhat larger change to ensure that all the `Registers` associated types lined up.